### PR TITLE
[MRG] Adding CI for documentation and fixing docs warnings

### PR DESCRIPTION
**Pull request by betatim**
- [x] Add / update documentation
 - [ ] Add tests

This adds a CircleCI job that builds the documentation to make sure it builds. It also turns all warnings into errors. The changes to the docs are there to fix the warnings that are now errors.

### CODE REVIEW

File: docs/contributing/docs.rst [hunk 1]
```diff
@@ -50,18 +50,18 @@ __ http://docutils.sourceforge.net/
...
 
 Then from the ``docs`` directory, build the HTML:
 
-.. console::
+.. code-block:: console
 
      $ make html
 
-To get started contributing, you'll want to read the :ref:`reStructuredText
-reference <sphinx:rst-index>`.
```
Comment (betatim) [inline docs/contributing/docs.rst]: What kind of reference was this meant to be? Something that goes straight to the sphinx documentation?

File: docs/index.rst [hunk 2]
```diff
@@ -30,7 +30,7 @@ Ubuntu 18.04. We have a bunch of tutorials to get you started.
      tutorials/jetstream
      tutorials/google
 
-- :ref:`tutorials/custom`.
+- :ref:`custom <tutorials/custom>`.
```
Comment (betatim) [inline docs/index.rst]: This still leads to an error:
```
/root/repo/docs/index.rst:33:undefined label: tutorials/custom (if the link has no caption the label must precede a section header)
```

I don't understand why :(

File: docs/contributing/docs.rst (hunk 1 above)
Comment (yuvipanda) [reply docs/contributing/docs.rst]: Yes, it is! I'm not sure why this isn't working. I copied this from the djago docs.

Comment (betatim) [reply docs/contributing/docs.rst]: Maybe it requires some kind of setup in `conf.py`? Can you link me to the source of the django docs, then I could investigate.

Comment (betatim) [reply docs/contributing/docs.rst]: Fixed.

**Comment by betatim** [comment]
Pending circle this is ready to be merged. Probably best to merge before other PRs to avoid having to catch up with new docs breakage that is introduced.

**Comment by yuvipanda** [comment]
<3 thanks, @betatim

**Pull request merged**